### PR TITLE
Add ResNet training demo doc

### PR DESCRIPTION
**Pull request by Yancey1989**

### CODE REVIEW

File: example/resnet/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+$IMAGENET_HOME/validation.tgz 2>&1 > resnet_train.log &
+```
+
+We can see the loss value is decreasing as the following screenshot.
+
+![loss](loss.jpg)
+
+## Benchmark with PyTorch
+
+To compare with PyTorch, you can also run PyTorch example script: [resnet.py](/example/resnet/resnet.py).
```
Comment (QiJune) [inline example/resnet/README.md:90]: Please add the benchmark environment? CPU or GPU? P100 or V100?

File: example/resnet/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# ResNet Training Demo
+
+This demo implements training the ResNet model on the ImageNet dataset.
+
+## Prepare ImageNet Data
+
+Because of the ImageNet dataset is not a public one, we should register
```
Comment (wangkuiyi) [inline example/resnet/README.md:7]: How big is the dataset? in GB, and in the number of images.

File: example/resnet/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+on the [ImageNet website](http://www.image-net.org/) then download the training
+and validation tarball from the [download site](http://www.image-net.org/challenges/LSVRC/2010/downloads).
+
+After finishing the downloading, you can create and export a home directory
```
Comment (wangkuiyi) [inline example/resnet/README.md:11]: What do we get from the downloading? A file or some files?

File: example/resnet/README.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+to save the ImageNet data.
+
+``` bash
+mkdir -p ~/imagenet
+export IMAGENET_HOME=~/imagenet
+```
+
+The following command extracts the tarball `ILSVRC2010_images_train.tar`
```
Comment (wangkuiyi) [inline example/resnet/README.md:19]: Is `ILSVRC2010_images_train.tar` one of the downloaded file?

**Review by wangkuiyi — APPROVED** [review]

**Review by wangkuiyi — APPROVED** [review]
